update-common/src/artifacts/update_plan.rs

```diff
@@ -317,6 +321,50 @@ impl<'a> UpdatePlanBuilder<'a> {
             },
         )?;
 
+        // We need to get all the signing information now to properly check
+        // version at builder time (builder time is not async)
+        let image_a = rot_a_data
+            .reader_stream()
+            .and_then(|stream| async {
```

Comment:
Could this use a `.await` rather than a future combinator? In general, `.await` is preferable -- it's easier to understand, particularly because `.and_then` exists both on futures and on results. (There are exceptions where future combinators make sense, but this doesn't seem to be one of them.)

Reply:
I was going off of what wicketd was doing (https://github.com/oxidecomputer/omicron/blob/main/wicketd/src/update_tracker.rs#L1679-L1696) but I'm fine with changing it for readability.